python/ctsm/modify_fsurdat/modify_fsurdat.py

```diff
@@ -0,0 +1,292 @@
+        summary = 'Modified fsurdat file'
+        contact = 'N/A'
+        data_script = os.path.abspath(__file__) + " -- " + get_git_sha()
+        description = 'Modified this file: ' + fsurdat_in
+        update_metadata(self.file, title=title, summary=summary,
+                        contact=contact, data_script=data_script,
+                        description=description)
+
+        # abort if output file already exists
+        file_exists = os.path.exists(fsurdat_out)
+        if file_exists:
+            errmsg = 'Output file already exists: ' + fsurdat_out
+            abort(errmsg)
+
+        # mode 'w' overwrites file if it exists
+        self.file.to_netcdf(path=fsurdat_out, mode='w',
+                            format="NETCDF3_64BIT")
+        logger.info('Successfully created fsurdat_out: %s', fsurdat_out)
+        self.file.close()
+
+
+    def set_dom_nat_pft(self, dom_nat_pft, lai, sai, hgt_top, hgt_bot):
+        """
+        Description
+        -----------
+        In rectangle selected by user (or default -90 to 90 and 0 to 360),
+        replace fsurdat file's PCT_NAT_PFT with:
+        - 100 for dom_nat_pft selected by user
+        - 0 for all other non-crop PFTs
+        If user has specified lai, sai, hgt_top, hgt_bot, replace these with
+        values selected by the user for dom_nat_pft
+
+        Arguments
+        ---------
+        dom_nat_pft:
+            (int) User's entry of PFT to be set to 100% everywhere
+        lai:
+            (float) User's entry of MONTHLY_LAI for their dom_nat_pft
+        sai:
+            (float) User's entry of MONTHLY_SAI for their dom_nat_pft
+        hgt_top:
+            (float) User's entry of MONTHLY_HEIGHT_TOP for their dom_nat_pft
+        hgt_bot:
+            (float) User's entry of MONTHLY_HEIGHT_BOT for their dom_nat_pft
+        """
+
+        for pft in self.file.natpft:
+            # initialize 3D variable; set outside the loop below
+            self.file['PCT_NAT_PFT'][pft,:,:] = \
+             self.file['PCT_NAT_PFT'][pft,:,:]. \
+                  where(self.not_rectangle, other=0)
+
+            var_name='MONTHLY_LAI'
+            if lai is not None:
+                self.set_lai_sai_hgts(pft=pft, dom_nat_pft=dom_nat_pft,
+                                      var_name=var_name, var=lai)
+            var_name='MONTHLY_SAI'
+            if sai is not None:
+                self.set_lai_sai_hgts(pft=pft, dom_nat_pft=dom_nat_pft,
+                                      var_name=var_name, var=sai)
+            var_name='MONTHLY_HEIGHT_TOP'
+            if hgt_top is not None:
+                self.set_lai_sai_hgts(pft=pft, dom_nat_pft=dom_nat_pft,
+                                      var_name=var_name, var=hgt_top)
+            var_name='MONTHLY_HEIGHT_BOT'
+            if hgt_bot is not None:
+                self.set_lai_sai_hgts(pft=pft, dom_nat_pft=dom_nat_pft,
+                                      var_name=var_name, var=hgt_bot)
```

Comment:
- [ ] Looking at the code here and in `set_lai_sai_hgts`, it looks unnecessary to have these calls to `set_lai_sai_hgts` in the `for pft` loop: you can take these out of the loop and just pass in `dom_nat_pft`, then getting rid of the `if pft == dom_nat_pft` conditional in `set_lai_sai_hgts`.

Reply:
Good catch!